virttest/virt_vm.py

```diff
@@ -654,21 +655,19 @@ def get_address(self, index=0, ip_version="ipv4"):
         will traverses from first nic to first element toward the end
         util get a reachable IP address;
         """
-        flexible_nic_index = ("yes" == self.params.get("flexible_nic_index", "no"))
-        macs, nics_index = self.virtnet.mac_list(), [index]
-        flexible_nic_index &= bool(len(macs) > 1)
-        if flexible_nic_index:
-            nics_index += [i for i in xrange(len(macs)) if i != index]
+        nr_nics = len(self.virtnet.mac_list())
+        nics_index = [index]
+        flexible = self.params.get("flexible_nic_index") == "yes"
+        if (flexible and nr_nics > 1):
+            nics_index += [i for i in xrange(nr_nics) if i != index]
+
         for index in nics_index:
             try:
                 return self._get_address(index, ip_version)
-            except VMAddressError:
-                if not flexible_nic_index:
+            except (VMMACAddressMissingError, VMIPAddressMissingError,
+                    VMAddressVerificationError):
+                if index == nics_index[-1]:
```

Comment:
Don't this variable `index` assigned in L664 conflict with `index` in function parameter `get_address(self, index=0, ip_version="ipv4")`

Reply:
It does look nasty, will fix this, thanks.